### PR TITLE
Reponse for internal calls

### PR DESCRIPTION
**Pull request by macaqueee**
Includes:

- ResponseEntity for internal calls

### CODE REVIEW

File: shared/model/response.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export default class ResponseEntity<T> {
+    success: boolean;
+    errors: Array<any>;
```
Comment (Verbalist) [inline shared/model/response.ts]: errors?:

File: shared/model/response.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    data: T;
```
Comment (Verbalist) [inline shared/model/response.ts]: data?:

File: shared/model/response.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+    constructor(data: T, errors: Array<any>) {
```
Comment (Verbalist) [inline shared/model/response.ts]: Array<string>

Comment (alex-arugon) [reply shared/model/response.ts]: data?

Comment (alex-arugon) [reply shared/model/response.ts]: how will we create response without `data`?

Comment (Verbalist) [reply shared/model/response.ts]: I think constructor is redundant. We can use it class only as type.

**Pull request merged**